watcher/tool/watcher/main.go

```diff
@@ -46,39 +62,80 @@ func run() error {
 	return nil
 }
 
-// waitForGrafana spins until Grafana API can be reached successfully or the provided context is cancelled
-func waitForGrafana(ctx context.Context, grafanaClient *lib.GrafanaClient) error {
+func runRollupsWatcher() error {
+	kubernetesClient, err := lib.NewKubernetesClient()
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	influxdbClient, err := lib.NewInfluxdbClient()
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	err = lib.WaitForAPI(context.TODO(), influxdbClient)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	ch, err := kubernetesClient.WatchConfigMaps(context.TODO(), lib.RollupsPrefix)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	receiveAndCreateRollups(context.TODO(), influxdbClient, ch)
+	return nil
+}
+
+// receiveAndCreateDashboards listens on the provided channel that receives new dashboards data and creates
+// them in Grafana using the provided client
+func receiveAndCreateDashboards(ctx context.Context, client *lib.GrafanaClient, ch <-chan string) {
 	for {
 		select {
-		case <-time.After(pollInterval):
-			err := grafanaClient.Health()
+		case data, ok := <-ch:
+			if !ok {
+				log.Warningf("dashboards channel closed")
+				return
+			}
+
+			err := client.CreateDashboard(data)
 			if err != nil {
-				log.Infof("cannot reach grafana: %v", trace.DebugReport(err))
-			} else {
-				return nil
+				log.Errorf("failed to create dashboard: %v", trace.DebugReport(err))
 			}
 		case <-ctx.Done():
-			return trace.Errorf("failed to reach grafana")
+			log.Infof("stopping")
+			return
 		}
 	}
 }
 
-// receiveAndCreateDashboards listens on the provided channel that receives new dashboards data and creates
-// them in Grafana using the provided client
-func receiveAndCreateDashboards(ctx context.Context, grafanaClient *lib.GrafanaClient, ch <-chan string) {
+// receiveAndCreateRollups listens on the provided channel that receives new rollups data and creates
+// them in Influxdb using the provided client
+func receiveAndCreateRollups(ctx context.Context, client *lib.InfluxdbClient, ch <-chan string) {
 	for {
 		select {
-		case data := <-ch:
-			err := grafanaClient.CreateDashboard(data)
+		case data, ok := <-ch:
+			if !ok {
```

Comment:
This looks pretty similar to `receiveAndCreateDashboards` - you could have a single function to abstract the receiver and only factor the handling of data into a separate function